src/plugins/usage_collection/server/collector/collector.ts

```diff
@@ -34,20 +34,20 @@ export interface SchemaField {
   type: string;
 }
 
-type Purify<T extends string> = { [P in T]: T }[T];
+export type RecursiveMakeSchemaFrom<U> = U extends object
+  ? MakeSchemaFrom<U>
+  : { type: AllowedSchemaTypes };
 
 export type MakeSchemaFrom<Base> = {
-  [Key in Purify<Extract<keyof Base, string>>]: Base[Key] extends Array<infer U>
-    ? { type: AllowedSchemaTypes }
-    : Base[Key] extends object
-    ? MakeSchemaFrom<Base[Key]>
-    : { type: AllowedSchemaTypes };
+  [Key in keyof Base]: Base[Key] extends Array<infer U>
```

Comment:
The test for verifying arrays of objects in the schema pass without the changes to the types but the changes make typescript itself stop complaining.